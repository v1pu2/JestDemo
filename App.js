/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';

import {StyleSheet, Text, View} from 'react-native';
import Home from './src/Home';

const App = () => {
  const check = n => {
    return n * 3;
  };
  return (
    <View style={styles.main}>
      <Text style={styles.text}>{check(6)}</Text>
      <Home />
    </View>
  );
};

const styles = StyleSheet.create({
  main: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: 'black',
    fontSize: 20,
  },
});

export default App;
