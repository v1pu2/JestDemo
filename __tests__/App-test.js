/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// it('renders correctly', () => {
//   renderer.create(<App />);
// });
test('App snapshot', () => {
  const snap = renderer.create(<App />).toJSON();
  expect(snap).toMatchSnapshot();
});

test('answer', () => {
  expect(3 * 3).toEqual(9)
});
