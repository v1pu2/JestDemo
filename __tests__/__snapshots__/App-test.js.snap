// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 20,
      }
    }
  >
    18
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
        }
      }
    >
      Home 
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter UserName"
      style={
        Object {
          "backgroundColor": "grey",
          "marginTop": 20,
          "padding": 10,
        }
      }
      testID="userName"
    />
  </View>
</View>
`;
